src/main/java/org/openrewrite/java/testing/jmockit/JMockitMockUpToMockito.java

```diff
@@ -0,0 +1,230 @@
+                        .map(JavaType.Method::getName)
+                        .collect(Collectors.toSet());
+
+                // Static
+                List<Statement> methods = mockedMethods
+                        .stream()
+                        .filter(s -> staticMethods.contains(((J.MethodDeclaration) s).getSimpleName()))
+                        .collect(toList());
+                if (!methods.isEmpty()) {
+                    mockStatements.append("MockedStatic<").append(className).append("> mockStatic").append(mockName).append(" = mockStatic(").append(className).append(".class);");
+                    for (Statement method : methods) {
+                        J.MethodDeclaration m = (J.MethodDeclaration) method;
+                        mockStatements.append("mockStatic").append(mockName).append(".when(() -> ").append(className).append(".").append(m.getSimpleName());
+                        appendMethod(mockStatements, m);
+                    }
+                    maybeAddImport("org.mockito.MockedStatic");
+                    shouldClose.add("mockStatic" + mockName);
+                }
+
+                // Instance
+                methods = mockedMethods
+                        .stream()
+                        .filter(s -> !staticMethods.contains(((J.MethodDeclaration) s).getSimpleName()))
+                        .collect(toList());
+                if (!methods.isEmpty()) {
+                    mockStatements.append("MockedConstruction<").append(className).append("> mockObj").append(mockName).append(" = mockConstruction(").append(className).append(".class, (mock, context) -> {");
+                    for (Statement method : methods) {
+                        J.MethodDeclaration m = (J.MethodDeclaration) method;
+                        mockStatements.append("when(mock.").append(m.getSimpleName());
+                        appendMethod(mockStatements, m);
+                    }
+                    mockStatements.append("});");
+
+                    maybeAddImport("org.mockito.MockedConstruction", null, false);
+                    shouldClose.add("mockObj" + mockName);
+                }
+
+                StringBuilder otherStatements = new StringBuilder();
+                newClass.getBody()
+                        .getStatements()
+                        .stream()
+                        .filter(s -> {
+                            if (s instanceof J.MethodDeclaration) {
+                                return ((J.MethodDeclaration) s).getLeadingAnnotations().stream()
+                                        .noneMatch(o -> TypeUtils.isOfClassType(o.getType(), "mockit.Mock"));
+                            }
+                            return true;
+                        })
+                        .forEach(s -> {
+                            otherStatements.append(s.printTrimmed());
+                            otherStatements.append(";");
+                        });
```

Comment:
This construct surprised me; we don't typically print existing statements back into a  JavaTemplate string. Is there a way we could do this retaining the original statements through either `ListUtils.flatMap` in something like this above instead of the for loop? `md.withBody(md.getBody().withStatements(ListUtils.flatMap(md.getBody().getStatements(), ....)))`

Reply:
Well, let's me try fixing it